src/components/Countries.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { fetchCountries } from '../redux/countries/countriesSlice';
+import { selectSearchTerm } from '../redux/searchbox/searchboxSlice';
+import styles from './countries.module.css';
+
+function Countries() {
+  const dispatch = useDispatch();
+
+  // Selectors
+  const countries = useSelector((state) => state.countries.countries);
+  const loading = useSelector((state) => state.countries.loading);
+  const error = useSelector((state) => state.countries.error);
+  const searchTerm = useSelector(selectSearchTerm);
+
+  useEffect(() => {
+    dispatch(fetchCountries());
+  }, [dispatch]);
+
+  // Filter countries based on the search term
+  const filteredCountries = countries.filter(
+    (country) => country.name.common.toLowerCase().includes(searchTerm.toLowerCase()),
+  );
+
+  // Render loading state
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  // Render error state
+  if (error) {
+    return <div>{error.message}</div>;
+  }
+
+  return (
+    <div className={styles.container}>
+      <h1 className={styles.header}>List of Countries</h1>
+      <div className={styles.countriesContainer}>
+        {filteredCountries.map((country) => (
+          <div className={styles.countryContainer} key={country.name.common}>
+            <img src={country.flags.svg} alt={country.flags.alt} className={styles.flag} />
+            <span className={styles.commonName}>
+              Common Name:
+              {' '}
+              {country.name.common}
+            </span>
+            <span className={styles.capitalName}>
+              Capital Name:
+              {' '}
+              {country.capital}
+            </span>
+            <span className={styles.population}>
+              Population:
+              {' '}
+              {country.population}
+            </span>
+            <Link to={`/country/${country.name.common}`} className={styles.link}>
+              More info
+            </Link>
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+}
+
+export default Countries;
```

Comment:
- [ ] Great job so far. However, there're few things to work on before you could move on.
- [ ] The main purpose of this project is to design a mobile application, its desktop version is a plus according to the project requirement. :point_down: 
- [ ] Number one, at this moment, your app is not responsive on the mobile.
- [ ] Number two, your'e required to follow the original design, this includes 
    -  Colors (select one main color for your website).
    -  Typography: font face, size and weight.
    -  Layout: composition and space between elements.

### Project Requirement
![project-rqt](https://github.com/ethred/Countries_Basic_information/assets/26597432/d45dc92d-d073-484f-9d41-10ab96df6652)

![project-rqt-2](https://github.com/ethred/Countries_Basic_information/assets/26597432/22b1adae-c96d-4036-8346-1479b0e6e9de)

- [ ] Number three, you're to design mobile application and make it to look like how it looks on the design. Your design does not look like original design. Kindly work on that too.

### Original Design
![temp-design](https://github.com/ethred/Countries_Basic_information/assets/26597432/ed5fb050-6bca-4509-b37a-8a8a2a3aeed3)

### Your Design

![your-design](https://github.com/ethred/Countries_Basic_information/assets/26597432/39f946a0-f959-41f0-b85c-314bfc3fedc4)

### App Behaviour
[app.webm](https://github.com/ethred/Countries_Basic_information/assets/26597432/0b2b6e9c-2dc2-456a-80af-5b95aaf628e0)